File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/CreatedBehaviourImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result of a create behaviour impulse
 */
class CreatedBehaviourImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * the behaviour instance
     */
    fun data(config: Behaviour.() -> Unit) = 
        add { "data{ " + Behaviour().apply(config).render(it) + "}"}

}

File: gaia-sdk-javascript/src/graphql/response/type/CreatedBehaviourImpulse.ts

import {Behaviour} from "./Behaviour";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../../request/enumeration/RuntimeState";
import {SkillState} from "../../request/enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderBy} from "../enumeration/OrderBy";

/**
* Impulse which indicates the result of a create behaviour impulse
*/
export interface CreatedBehaviourImpulse {
    id?:Uuid, 
    /**
    * the behaviour instance
    */
    data?:Behaviour
}